vita3k/gui/include/gui/imgui_impl_sdl_vulkan.h

```diff
@@ -0,0 +1,56 @@
+// Temporary Vulkan ImGui Implementation
+
+#pragma once
+
+#include <imgui.h>
+
+#include <gui/imgui_impl_sdl_state.h>
+#include <host/window.h>
+#include <renderer/vulkan/state.h>
+
+typedef union SDL_Event SDL_Event;
+
+struct ImGui_VulkanState : public ImGui_State {
+    vk::ShaderModule vertex_module;
+    vk::ShaderModule fragment_module;
+
+    vk::RenderPass renderpass;
+    vk::Framebuffer framebuffers[2];
+    vk::DescriptorSetLayout matrix_layout;
+    vk::DescriptorSetLayout sampler_layout;
+    vk::DescriptorPool descriptor_pool;
+    vk::DescriptorSet matrix_set;
+    vk::PipelineLayout pipeline_layout;
+    vk::Pipeline pipeline;
+
+    vk::Sampler sampler;
+    ImTextureID font_texture{};
+    VmaAllocation draw_allocation = VK_NULL_HANDLE;
+    vk::Buffer draw_buffer;
+    size_t draw_buffer_vertices = 0;
+    VmaAllocation index_allocation = VK_NULL_HANDLE;
+    vk::Buffer index_buffer;
+    size_t index_buffer_indices = 0;
+    VmaAllocation transformation_allocation = VK_NULL_HANDLE;
+    vk::Buffer transformation_buffer;
+
+    vk::CommandBuffer command_buffer;
+
+    vk::Semaphore image_acquired_semaphore;
+    vk::Semaphore render_complete_semaphore;
+
+    inline renderer::vulkan::VulkanState &get_renderer() {
+        return dynamic_cast<renderer::vulkan::VulkanState &>(*renderer);
```

Comment:
static if you can. Dynamic cast is not really good, but not really bad either

Reply:
I'll make it static (although I like the whole get style like this like in SceGxmProgram). What do you want me to use for cast instead?

Reply:
No i mean static cast